converter/converter.go

```diff
@@ -28,10 +29,13 @@ const (
 // because of mismatched functionality, an error is returned with no resulting
 // config. If the conversion completed successfully but generated warnings, an
 // error is returned alongside the resulting config.
-func Convert(in []byte, kind Input) ([]byte, error) {
+func Convert(in []byte, kind Input) ([]byte, common.Diagnostics) {
```

Comment:
This doesn't work, because it's a public API exposing an internal type, meaning callers don't have access to it.

The typical way of dealing with this is to either create aliases (see [river.ErrNoConversion](https://github.com/grafana/agent/blob/4b8807a517071556c7fd520d7ddd792fbcd311f7/pkg/river/types.go#L51)) or to create a lower-level type which is used internally, and to convert that to a higher-level public API error type.

Reply:
good call, should be fixed now